File: apps/merchant-web/src/api/withdrawAccount.ts
import request from '@/utils/request'
import type { ICommonResponse } from '@/types/common'

export interface IWithdrawAccount {
  id: number
  type: string // CRYPTO / BANK
  chain?: string
  address?: string
  bankName?: string
  accountNo?: string
  accountName?: string
  swiftCode?: string
  country?: string
  remark?: string
  isDefault?: boolean
  status?: string
  createdAt?: string
}

export const withdrawAccountApi = {
  list: () => request.get<ICommonResponse<IWithdrawAccount[]>>('/merchant/withdraw-accounts'),
  create: (data: Partial<IWithdrawAccount>) =>
    request.post<ICommonResponse<any>>('/merchant/withdraw-accounts', data),
  update: (id: number, data: Partial<IWithdrawAccount>) =>
    request.put<ICommonResponse<any>>('/merchant/withdraw-accounts/' + id, data),
  remove: (id: number) =>
    request.delete<ICommonResponse<any>>('/merchant/withdraw-accounts/' + id),
  setDefault: (id: number) =>
    request.put<ICommonResponse<any>>('/merchant/withdraw-accounts/' + id + '/default', {}),
}

File: apps/api/src/main/java/com/mall/api/modules/merchant/entity/WithdrawAccount.java
package com.mall.api.modules.merchant.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableField;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.io.Serializable;
import java.time.LocalDateTime;

@Data
@TableName("withdraw_account")
public class WithdrawAccount implements Serializable {

    private static final long serialVersionUID = 1L;

    @TableId(type = IdType.ASSIGN_ID)
    private Long id;

    @TableField("merchant_id")
    private Long merchantId;

    @TableField("user_id")
    private Long userId;

    /** CRYPTO / BANK */
    private String type;

    // 加密货币
    private String chain;
    private String address;

    // 银行
    @TableField("bank_name")
    private String bankName;

    @TableField("account_no")
    private String accountNo;

    @TableField("account_name")
    private String accountName;

    @TableField("swift_code")
    private String swiftCode;

    private String country;
    private String remark;

    @TableField("is_default")
    private Boolean isDefault;

    private String status;
    private Boolean deleted;

    @TableField("created_at")
    private LocalDateTime createdAt;

    @TableField("updated_at")
    private LocalDateTime updatedAt;
}
